onebusaway-gtfs-transformer/src/main/java/org/onebusaway/gtfs_transformer/impl/SanitizeStopIdForApiAccess.java

```diff
@@ -0,0 +1,44 @@
+/**
+ * Copyright (C) 2018 Cambridge Systematics, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.onebusaway.gtfs_transformer.impl;
+
+import org.onebusaway.gtfs.model.*;
+import org.onebusaway.gtfs.services.GtfsMutableRelationalDao;
+import org.onebusaway.gtfs_transformer.services.GtfsTransformStrategy;
+import org.onebusaway.gtfs_transformer.services.TransformContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collection;
+
+public class SanitizeStopIdForApiAccess implements GtfsTransformStrategy {
+    private final Logger _log = LoggerFactory.getLogger(SanitizeStopIdForApiAccess.class);
+    private final String sanitize_selector = "[\\[||\\:||\\]||\\@||\\.|| ||\\\\||\\(||\\)||\\_||\\-||\\/||\\\"]";
+
+    @Override
+    public String getName() {
+        return this.getClass().getSimpleName();
+    }
+
+    @Override
+    public void run(TransformContext context, GtfsMutableRelationalDao dao) {
+        _log.info("Removing the following characters from trip names: [, :, ], \\, (, ), _, @, and .");
+        Collection<Stop> Stops = dao.getAllStops();
+        for (Stop stop : Stops) {
```

Comment:
let's change this to only actually perform the set(...) if a match(...) is positive.  

See https://docs.oracle.com/javase/7/docs/api/java/util/regex/Pattern.html.

Basically under the covers a lot of needless work might be getting done by calling set even if the next result is zero.  (Happy to be proven wrong here BTW)

Reply:
No that's a good change-- in the very specific bus company I had in mind their naming convention includes those characters universally, but it really shouldn't set(...) each time